controllers/utils/manifestworkreplicaset.go

```diff
@@ -0,0 +1,227 @@
+		},
+	}
+	expectedValueAnn := fmt.Sprintf(manifestWorkExpectedValuesAnnotationTemplate, "isPrepCompleted")
+	return generateManifestWorkReplicaset(name, namespace, expectedValueAnn, jsonPaths, ibu)
+}
+
+func generateManifestWorkReplicaset(name, namespace, expectedValueAnn string, jsonPaths []mwv1.JsonPath, ibu *lcav1alpha1.ImageBasedUpgrade) (*mwv1alpha1.ManifestWorkReplicaSet, error) {
+	ibuRaw, err := ibuToBytes(ibu)
+	if err != nil {
+		return nil, err
+	}
+	manifestConfigs := []mwv1.ManifestConfigOption{
+		{
+			ResourceIdentifier: mwv1.ResourceIdentifier{
+				Name:     ibu.GetName(),
+				Group:    ibu.GroupVersionKind().Group,
+				Resource: ibu.GetResourceVersion(),
+			},
+			FeedbackRules: []mwv1.FeedbackRule{
+				{
+					Type:      mwv1.JSONPathsType,
+					JsonPaths: jsonPaths,
+				},
+			},
+		},
+	}
+	mwrs := &mwv1alpha1.ManifestWorkReplicaSet{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				manifestWorkExpectedValuesAnnotation: expectedValueAnn,
+			},
+		},
+		Spec: mwv1alpha1.ManifestWorkReplicaSetSpec{
+			PlacementRefs: []mwv1alpha1.LocalPlacementReference{
+				{
+					Name: "dummy",
+				},
+			},
+			ManifestWorkTemplate: mwv1.ManifestWorkSpec{
+				Workload: mwv1.ManifestsTemplate{
+					Manifests: []mwv1.Manifest{{RawExtension: runtime.RawExtension{Raw: ibuRaw}}},
+				},
+				DeleteOption:    &mwv1.DeleteOption{PropagationPolicy: mwv1.DeletePropagationPolicyTypeOrphan},
+				ManifestConfigs: manifestConfigs,
+			},
+		},
+	}
+	return mwrs, nil
+}
+
+func ibuToBytes(ibu *lcav1alpha1.ImageBasedUpgrade) ([]byte, error) {
+	scheme := runtime.NewScheme()
+	lcav1alpha1.AddToScheme(scheme)
+	s := serializer.NewSerializerWithOptions(serializer.DefaultMetaFactory, scheme, scheme, serializer.SerializerOptions{})
+	gvks, isUnversioned, err := scheme.ObjectKinds(ibu)
+	if err != nil {
+		return []byte{}, err
+	}
+	if !isUnversioned && len(gvks) == 1 {
+		ibu.TypeMeta = metav1.TypeMeta{
+			Kind:       gvks[0].Kind,
+			APIVersion: gvks[0].GroupVersion().Identifier(),
+		}
+	}
+	var dst bytes.Buffer
+	s.Encode(ibu, &dst)
+	return dst.Bytes(), nil
+}
+
+func getCGUNameForIBGU(ibgu *ibguv1alpha1.ImageBasedGroupUpgrade, templates []string) string {
+	s := []string{}
+	for _, template := range templates {
+		s = append(s, strings.ReplaceAll(template, ibgu.GetName()+"-", ""))
+	}
+	return fmt.Sprintf("%s-%s", ibgu.Name, strings.Join(s, "-"))
+}
+
+// GenerateClusterGroupUpgradeForIBGU returns a populated CGU for an IBGU
+func GenerateClusterGroupUpgradeForIBGU(ibgu *ibguv1alpha1.ImageBasedGroupUpgrade, templateNames, blockingCGUs []string) *ranv1alpha1.ClusterGroupUpgrade {
+	blockingCRs := []ranv1alpha1.BlockingCR{}
+	for _, cguName := range blockingCGUs {
+		blockingCRs = append(blockingCRs, ranv1alpha1.BlockingCR{
+			Name:      cguName,
+			Namespace: ibgu.GetNamespace(),
+		})
+	}
+	enable := true
+	return &ranv1alpha1.ClusterGroupUpgrade{
```

Comment:
We need to annotate the cluster with "import.open-cluster-management.io/disable-auto-import" annotation before starting upgrade and remove it after upgrade is done.
```
apiVersion: ran.openshift.io/v1alpha1
kind: ClusterGroupUpgrade
metadata:
  name: cgu-ibu-upgrade
  namespace: default
spec:
  actions:
    beforeEnable:
      addClusterAnnotations:
        import.open-cluster-management.io/disable-auto-import: "true" 
    afterCompletion:
      removeClusterAnnotations:
      - import.open-cluster-management.io/disable-auto-import 
```
If it's abort or rollback, we need to remove the annotation as well.

Reply:
updated